ncm-download/src/main/perl/download.pm

```diff
@@ -306,12 +308,25 @@ sub retrieve
                 if (!POSIX::WIFEXITED($?) || POSIX::WEXITSTATUS($?) != 0) {
                     # Only warn as there may be attempts from multiple proxies.
                     $self->warn("curl failed (". ($?>>8) ."): $errs\nCommand = $proc");
+                    if (-e $tmpfn) {
```

Comment:
can you use `CAF::Path`? so we can unittest it?
also, you should also do this before the download